### PR TITLE
workflows: use local scratch space

### PR DESCRIPTION
**Pull request by kaplun**
Reducing usage of EOS, when what is needed is only a temporary disk space.

- [x] I have all the information that I need (if not, move to `RFC` and look for it).
- [ ] I linked the related issue(s) in the corresponding commit logs.
- [x] I wrote [good commit log messages](https://github.com/torvalds/subsurface-for-dirk/blob/5f15ad5a86ada3c5e574041a5f9d85235322dabb/README#L92-L119).
- [x] My code follows the code style of this project.
- [ ] I've added any new docs if API/utils methods were added.
- [ ] I have updated the existing documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

Rather than using local .tar directory (currently on a networked
filesystem) as a scratch space for extracting plots and author.xml
file, uses a temporary local directory that is cleaned up at the end.

Signed-off-by: Samuele Kaplun <samuele.kaplun@cern.ch>

### CODE REVIEW

File: inspirehep/modules/workflows/tasks/arxiv.py [hunk 1]
```diff
@@ -122,24 +124,28 @@ def arxiv_plot_extract(obj, eng):
     tarball = obj.files[filename]
 
     if tarball:
+        scratch_space = mkdtemp('plot_extract')
```
Comment (kaplun) [inline inspirehep/modules/workflows/tasks/arxiv.py]: I discovered we can globally set what root directory to use :+1: by simply setting: 
```python
import tempfile
tempfile.tempdir = '/bla'
```

in the config file.

**Comment by kaplun** [comment]
cc: @jacquerie, @david-caro. Yay!

File: inspirehep/modules/workflows/tasks/arxiv.py (hunk 1 above)
Comment (david-caro) [reply inspirehep/modules/workflows/tasks/arxiv.py]: It also respects the env vars 'TMPDIR' and similar.

File: inspirehep/modules/workflows/tasks/arxiv.py [hunk 2]
```diff
@@ -122,24 +124,28 @@ def arxiv_plot_extract(obj, eng):
...
-            plots = process_tarball(tarball.file.uri)
-        except (InvalidTarball, NoTexFilesFound):
-            obj.log.error(
-                'Invalid tarball %s for arxiv_id %s', tarball.file.uri, arxiv_id)
-            return
-        except DelegateError as err:
-            obj.log.error('Error extracting plots for %s. Report and skip.', arxiv_id)
-            current_app.logger.exception(err)
-            return
+            try:
```
Comment (david-caro) [inline inspirehep/modules/workflows/tasks/arxiv.py]: This double try is ugly :/, maybe we can us ea context manager instead?

**Pull request closed without merging**